packages/loader/container-loader/src/container.ts

```diff
@@ -829,7 +829,7 @@ export class Container extends EventEmitterWithErrorHandling<IContainerEvents> i
                 }
 
                 // take summary and upload
-                const appSummary: ISummaryTree = this.context.createSummary(redirectTable);
+                const appSummary: ISummaryTree = await this.context.createSummary(redirectTable);
```

Comment:
Same problem here - await needs to be after this.emit("attached"). I'm not even sure code can handle it correctly today (i.e. async inflight summary process while we emit attached event) - maybe, but I'd need to do deeper code inspection to be sure, as we never have had such state.